### PR TITLE
ai: Add mirrored gini-doc feature documentation skill

### PR DESCRIPTION
**Pull request by ValentinaIancu-Gini**
## Pull Request Description

Adds the `/gini-doc` Claude skill for generating plain Markdown feature documentation pages for Confluence from GiniBankSDK code changes on the current branch, mirrored with `gini-mobile-android`.

The skill was originally added on this branch as `/generate-feature-docs` (iOS-only, monolithic). This PR renames and restructures it to the shared cross-repo shape already used by the Android repo:

- **`SKILL.md`** — platform-agnostic workflow, byte-identical to the Android copy (enforced by `shared-skills.check.yml` via `.github/mirrored-skills.txt`). Adds the `--platform` argument, base-branch detection, and template selection.
- **`platform.md`** — iOS-specific conventions (not mirrored): source roots, `de.lproj`/`en.lproj` localization files with `ginibank.`/`ginicapture.` key prefixes, `GiniBankConfiguration` configuration surface, OS-permission → `Info.plist` mapping, `[term:]` table and `[snippet:]` code blocks (Swift, one-parameter-per-line style).
- **`templates/`** — four mirrored templates derived from published documentation pages: `major-feature`, `flag-feature`, `os-integration`, `transfer-summary-extension`. Platform specifics resolve from `platform.md` at generation time.
- **`.github/mirrored-skills.txt`** — updated to match the Android copy, adding the `gini-doc` files to the mirror list.
- **`CLAUDE.md`** — skill listing updated to `/gini-doc --platform ios`.

Output path follows the shared convention: `docs/ios/features/[feature-slug]/[feature-slug].md`.

## Notes for Reviewers

- Verified `SKILL.md`, all four templates, and `mirrored-skills.txt` are byte-identical to `gini-mobile-android` (`diff` clean), so the shared-skills CI check passes.
- All API references in `platform.md` snippets were verified against source: `GiniBank.viewController(withClient:...)`, `GiniCaptureResultsDelegate` callbacks, `GiniBankConfiguration.shared.cleanup()`.
- Known follow-up: the `ui-customization-guide-url` term currently points to the IBSV space overview with a `TOBEADDED` marker — replace with the exact iOS UI Customisation Guide page once pinned.
- Documentation/tooling change only — no SDK source code touched, no tests affected.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .claude/skills/gini-doc/platform.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+`GiniBank.viewController(withClient:...)`.
+
+## OS permissions → Info.plist requirements
+
+Infer `Info.plist` requirements from OS APIs used in source:
+
+- `PHPhotoLibrary` (add-only access) → `NSPhotoLibraryAddUsageDescription`
+  required
+- `PHPickerViewController` / full photo library access →
+  `NSPhotoLibraryUsageDescription` required
```
[the remark this answers is not in the document]
Comment (zladzeyka) [reply .claude/skills/gini-doc/platform.md:64]: - Pre-iOS 14 (iOS 8–13): picking from the photo library meant UIImagePickerController with .photoLibrary source, which reads through PHPhotoLibrary. That path required NSPhotoLibraryUsageDescription. So back then, "use the photo picker → declare the key" was correct
  advice.   
  - iOS 14+ (2020): Apple introduced PHPickerViewController specifically to eliminate that requirement — it runs out-of-process, the user picks in a system UI, and only the selected items are returned. No permission dialog, no Info.plist key needed.

Means we don't need `NSPhotoLibraryUsageDescription` anymore and another TODO cleanup remove it from the integration guide

File: .claude/skills/gini-doc/platform.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+| Term | Value |
+|---|---|
+| `config-object` | `GiniBankConfiguration` |
+| `sdk-entry-point` | `GiniBank` |
+| `manifest-file` | `Info.plist` |
+| `result-type` | `GiniCaptureResultsDelegate` callback |
+| `success-result-case` | `giniCaptureAnalysisDidFinishWith(result:)` |
+| `empty-result-case` | `giniCaptureDidEnterManually()` (no-results path) |
+| `cleanup-call` | `GiniBankConfiguration.shared.cleanup()` |
```
Comment (zladzeyka) [inline .claude/skills/gini-doc/platform.md:100]: [Suggestion]: Could you please check if this correct reference for cleanup?

File: .claude/skills/gini-doc/platform.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    return configuration
+}()
+```
+
+### `result-callback`
+
+```swift
+extension YourViewController: GiniCaptureResultsDelegate {
+
+    /// Called when the analysis finished with results.
```
[the remark this answers is not in the document]
Comment (zladzeyka) [reply .claude/skills/gini-doc/platform.md]: [Suggestion]: please align the comment style here

File: .claude/skills/gini-doc/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+
+### Audience
+
+Documentation serves three audiences simultaneously:
+
+| Audience | What they need |
+|---|---|
+| **Developers** | Precise inputs, outputs, conditions, dependencies, edge cases, and implementation examples |
+| **QA engineers** | Expected behavior, validation rules, failure cases, and edge cases specific enough to derive test scenarios |
+| **Product Managers / Product Owners** | Business intent, user outcomes, workflow, and scope — without unnecessary implementation detail |
```
Comment (zladzeyka) [inline .claude/skills/gini-doc/SKILL.md:74]: [Suggestion]:  Please add customer support or customer experience team here as well

File: .claude/skills/gini-doc/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+  .claude/skills/gini-doc/SKILL.md in gini-mobile-ios.
+  If you change it here, open a paired PR in the other repo with the same
+  content. CI (shared-skills.check.yml) fails when the copies diverge.
+  Platform-specific rules do NOT belong here — they live in the sibling
+  platform.md, which is intentionally different per repo.
+-->
+
+# Skill: /gini-doc
+
+Generate a plain Markdown (`.md`) feature documentation page for Confluence from GiniBankSDK source code changes on the current branch.
```
Comment (zladzeyka) [inline .claude/skills/gini-doc/SKILL.md:18]: [Clarification]: is `plain Markdown` good enough for our Confluence?

**Review by zladzeyka — CHANGES_REQUESTED** [review]
Thank you for PR🙏

File: .claude/skills/gini-doc/platform.md (hunk 2 above)
Comment (qnaveed87) [reply .claude/skills/gini-doc/platform.md:100]: The gallery picker on our side is AlbumsPickerViewController + direct PHPhotoLibrary.requestAuthorization(for: .readWrite) that uses NSPhotoLibraryUsageDescription

Comment (qnaveed87) [reply .claude/skills/gini-doc/platform.md:100]: Verified — GiniBankConfiguration.shared.cleanup() is real, at GiniBankConfiguration.swift

File: .claude/skills/gini-doc/SKILL.md (hunk 5 above)
Comment (qnaveed87) [reply .claude/skills/gini-doc/SKILL.md:18]: Yes , Confluence accepts Markdown

File: .claude/skills/gini-doc/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+
+The valid value for this repository is defined in `platform.md` (in this skill's directory).
+
+Retained as a required argument to keep the skill interface consistent with other Gini repositories.
+
+### `--feature-slug`
+
+Kebab-case name for the feature. Used as the output file name.
+
+Example: `cross-border-payments` → output file `cross-border-payments.md`
```
[the remark this answers is not in the document]
Comment (qnaveed87) [reply .claude/skills/gini-doc/SKILL.md:48]: The docs/-boundary rule at line 260 already covers the risk, this is a locally-invoked developer skill, and adding a regex would be redundant documentation weight

File: .claude/skills/gini-doc/templates/transfer-summary-extension.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+> **Note:** To use the [Feature Name] feature, please contact Gini Customer Support to have it enabled in our backend platform.
+
+> **Info:** We highly recommend having a QA session with Gini before releasing the [Feature Name] feature to your customers.
+
+[One paragraph: what triggers the new extraction — document content, QR code data, user action — and what the user experiences in the integrating app.]
+
+You will see the [fieldName] extraction inside the specific extractions of `[term: success-result-case]`:
+
+```json
+[Example extraction payload showing the new field among existing ones, copied from verified source or the Gini Bank API documentation. Mark the new field with a comment.]
```
[the remark this answers is not in the document]
Comment (qnaveed87) [reply .claude/skills/gini-doc/templates/transfer-summary-extension.md]: Done

File: .claude/skills/gini-doc/platform.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (qnaveed87) [reply .claude/skills/gini-doc/platform.md]: Done

**Review by zladzeyka — APPROVED** [review]
